src/components/CategoryCard/index.tsx

```diff
@@ -1,13 +1,33 @@
-import React from 'react';
+import React, { FC, useCallback, memo } from 'react';
 import { TouchableOpacity } from 'react-native';
-import { CategoryCardContainer, CategoryCardName } from './styles';
+import {
+  CategoryCardContainer,
+  CategoryCardName,
+  SelectedCategoryCardContainer,
+} from './styles';
+import { Props } from './types';
+
+export const CategoryCard: FC<Props> = ({ category, filter, setFilter }) => {
+  const handleFilter = useCallback(
+    () => setFilter(category),
+    [category, setFilter],
+  );
+
+  const isFiltered = category.localeCompare(filter) === 0;
```

Comment:
Don't use localCompare. It's **never** necessary on typescript.

```suggestion
  const isFiltered = category === filter;
```